File: solutions/1302-delete-characters-to-make-fancy-string/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var makeFancyString = function(s) {
    let output = "";
    
    let consecutives = 0;
    let prev = "X"
    
    for(const char of s){
        if(char !== prev){
            consecutives = 1;
            prev = char;
            output += char;
        }else{
            consecutives += 1;
            if(consecutives < 3){
                output += char;
            }
        }
    }
    
    return output;
};

File: solutions/1302-delete-characters-to-make-fancy-string/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        out = []

        for c in s:
            if len(out) >= 2 and c == out[-1] and c == out[-2]:
                continue
            else:
                out.append(c)

        return "".join(out)
